### PR TITLE
feat: Markdown

### PR DESCRIPTION
**Pull request by shivani170**
# Description

Markdown

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [x] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@devtron-labs/devtron-fe-common-lib",
-    "version": "0.0.74",
```
Comment (eshankvaish) [inline package.json:3]: pls check the correct version

File: package.json [hunk 2]
```diff
@@ -77,7 +77,9 @@
         "react-router": "^5.3.0",
         "react-router-dom": "^5.3.0",
         "react-draggable": "^4.4.5",
-        "yaml": "^2.4.1"
+        "yaml": "^2.4.1",
+        "marked": "4.3.0",
+        "dompurify": "^3.0.2"
```
Comment (eshankvaish) [inline package.json:82]: we would need these in dev dependencies as well

File: src/Common/Helper.tsx [hunk 3]
```diff
@@ -369,7 +369,7 @@ export function useJsonYaml(value, tabSize = 4, language = 'json', shouldRun = f
         }
         if (obj && typeof obj === 'object') {
             setJson(JSON.stringify(obj, null, tabSize))
-            setYaml(YAML.stringify(obj, { indent: 2, lineWidth: 0 }))
+            setYaml(YAML.stringify(obj, { indent: 2 }))
```
Comment (eshankvaish) [inline src/Common/Helper.tsx:372]: any specific reason for removing this?

File: src/Common/CodeEditor/CodeEditor.tsx [hunk 4]
```diff
@@ -300,7 +300,7 @@ const CodeEditor: React.FC<CodeEditorInterface> & CodeEditorComposition = React.
             }
             let final = value
             if (obj) {
-                final = state.mode === 'json' ? JSON.stringify(obj, null, tabSize) : YAMLStringify(obj, { indent: 2 })
+                final = state.mode === 'json' ? JSON.stringify(obj, null, tabSize) : YAML.stringify(obj, { indent: 2 })
```
Comment (eshankvaish) [inline src/Common/CodeEditor/CodeEditor.tsx:303]: FYI this code editor is not exported at present. Also shouldn't we update the dashboard as well

File: src/Common/Markdown/MarkDown.tsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
+import { marked } from 'marked'
+import DOMPurify from 'dompurify'
+import { useEffect, useRef } from 'react'
+
+const MarkDown = (
```
Comment (eshankvaish) [inline src/Common/Markdown/MarkDown.tsx:5]: If migrated from dashboard, please mark the one in dashboard as deprecated

File: src/Common/Markdown/MarkDown.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    setExpandableIcon: (arg0: boolean) => void,
```
Comment (eshankvaish) [inline src/Common/Markdown/MarkDown.tsx:6]: better name for arg0

File: src/Common/Markdown/MarkDown.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    markdown = '',
+    className = '',
+    breaks = false,
+    disableEscapedText = false,
+    ...props: any[]
```
Comment (eshankvaish) [inline src/Common/Markdown/MarkDown.tsx:11]: - add types, also create a separate file for types
- We are not exporting the component

File: src/Common/Markdown/MarkDown.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+) => {
+    const uncheckedCheckboxInputElement = `<input checked="" disabled="" type="checkbox">`
+    const checkedCheckboxInputElement = `<input disabled="" type="checkbox">`
+    const renderer = new marked.Renderer()
```
Comment (eshankvaish) [inline src/Common/Markdown/MarkDown.tsx:15]: all three can be moved outside the component

File: src/Common/Markdown/MarkDown.tsx [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    useEffect(() => {
+        getHeight()
+    }, [markdown])
+
+    function isReadmeInputCheckbox(text: string) {
+        if (text.includes(uncheckedCheckboxInputElement) || text.includes(checkedCheckboxInputElement)) {
+            return true
+        }
+        return false
+    }
```
Comment (eshankvaish) [inline src/Common/Markdown/MarkDown.tsx:35]: Also, we can use arrow function

```suggestion
    function isReadmeInputCheckbox(text: string) {
        return (text.includes(uncheckedCheckboxInputElement) || text.includes(checkedCheckboxInputElement))
    }
```

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Common/CodeEditor/CodeEditor.tsx (hunk 4 above)
Comment (eshankvaish) [reply src/Common/CodeEditor/CodeEditor.tsx]: leave a todo

File: src/Common/Markdown/MarkDown.tsx (hunk 9 above)
Comment (eshankvaish) [reply src/Common/Markdown/MarkDown.tsx]: .

**Review by eshankvaish — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Comment by shivani170** [comment]
Merged in demo/release

**Pull request closed without merging**